### PR TITLE
Add var operation to direct bindings

### PR DESCRIPTION
**Pull request by rdspring1**
This PR also adds `test_right_shift_logical` and `test_right_shift_logical_sizeof_dtype`, which didn't need any additional changes.

### CODE REVIEW

**Comment by rdspring1** [comment]
!test

File: tests/python/direct/test_python_frontend.py [hunk 1]
```diff
@@ -2315,3 +2338,99 @@ def fusion_func(fd: FusionDefinition) -> None:
...
+        ],
+        device="cuda",
+    )
+
+    for idx, dtype in enumerate(dtypes):
+        current_input = input.to(dtype)
+        num_bits = 32 if (dtype == torch.int32) else 64
+
+        # expected_outputs given by jax.lax.shift_right_logical(inputs, sizeof(dtype))
+        expected_output = torch.zeros_like(current_input)
```
Comment (jjsjann123) [inline tests/python/direct/test_python_frontend.py:2429]: what's happening here? looks like we are just creating zeros as reference here?

Comment (rdspring1) [reply tests/python/direct/test_python_frontend.py:2429]: When you right-shift the input by 32 or 64, the output is all zeros. 
🤷🏼 I don't remember why we have this test.
```python
>>> jax.lax.shift_right_logical(input.to('cpu').numpy(), 32)
Array([0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)
>>> jax.lax.shift_right_logical(input.to('cpu').numpy(), 64)
Array([0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)
```

**Comment by rdspring1** [comment]
!test

File: tests/python/direct/test_python_frontend.py (hunk 1 above)
Comment (jjsjann123) [reply tests/python/direct/test_python_frontend.py:2429]: 😃 that test doesn't make sense to my monkey brain. Can we change it to actually test some sane behavior?

**Review by jjsjann123 — APPROVED** [review]
I'm stamping to unblock.

but I hated that test so much and would really appreciate that we can change it.

File: tests/python/direct/test_python_frontend.py (hunk 1 above)
Comment (rdspring1) [reply tests/python/direct/test_python_frontend.py:2429]:
```cpp
// The logical_right_shift operation shifts the value's bits to the right.
// If the value is negative, it appends zeros to the front of the value.
// The sign is preserved with arithmetic_right_shift, so ones are pushed to the
// most significant bit.
```
IIUC, it tests that zeros are prepended to the shifted value. For `arithmetic_right_shift`, the shifted value is the signbit. e.g., `shift_right_logical(-1, 32) == 0` and `shift_right_arithmetic(-1, 32) == -1`

Comment (jjsjann123) [reply tests/python/direct/test_python_frontend.py:2429]: I'm not saying the behavior is not correct. But that feels like we are testing *only* a corner case.

i.e. this feels to me like adding an `add` and the only test you put there is `add( a, 'inf') == 'inf'` without having a `add(a, b) == (a + b)`

Comment (rdspring1) [reply tests/python/direct/test_python_frontend.py:2429]: `test_right_shift_bitwise` tests `jax.lax.shift_right_logical(inputs, 3)`.

Comment (rdspring1) [reply tests/python/direct/test_python_frontend.py:2429]: It should be `test_right_shift_logical`.

**Comment by rdspring1** [comment]
!build

File: tests/python/direct/test_python_frontend.py (hunk 1 above)
Comment (jjsjann123) [reply tests/python/direct/test_python_frontend.py:2429]: ahahah, sorry I was totally blind. Somehow I forgot about the the other test.

🤕

Comment (jjsjann123) [reply tests/python/direct/test_python_frontend.py:2429]: Thanks for bearing with me. 🙇
